Спринт 22

il76 opened the pull request:
Я так и не смог нормально стилизовать поля ввода на экране создания плейлиста через состояния.
В PlaylistAddFragment лежит закомментированная попытка покрасить бордеры программным способом, но, к сожалению, при потере фокуса это состояние пропадает.

Commit:
Правила игнора.
Версии компонентов.

Commit:
Экспериментируем с BottomSheet

Commit:
Промежуточные файлы по добавлению плейлиста

Commit:
Экран добавления плейлиста

Commit:
Починил открытие экрана нового плейлиста

Commit:
Куски для плейлистов

Commit:
Поведение BottomSheet.
Пару из плейлиста и трека заменил на датакласс PlaylistTrack

Commit:
Запрашиваем подтверждение при выходе из экрана создания плейлиста. Сохранение изображения в приватное хранилище

Commit:
Ещё классы для работы с плейлистами

Commit:
Ещё классы для работы с плейлистами

Commit:
DI

Commit:
Сохранение плейлиста

Commit:
Вывод плейлистов в медиатеке

Commit:
Вывод списка плейлистов по дизайну

Commit:
Правильный вывод списков плейлистов на странице плеера и на общем списке плейлистов

Commit:
Добавление трека в плейлист

Commit:
Возвращаем счётчик треков при запросе списка плейлистов

Commit:
При выходе с экрана создания плейлиста добавляем проверку на заполненные поля

Commit:
Эксперименты с вёрсткой страницы добавления плейлиста

Commit:
Состояние для кнопки

Commit:
Ещё попытки покрасить поля

IliaAnK commented on app/src/main/java/com/il76/playlistmaker/media/data/db/dao/PlaylistDao.kt:
```diff
@@ -0,0 +1,26 @@
+        FROM playlists p
+        ORDER BY p.id DESC
+    """)
+    suspend fun getPlaylists(): List<PlaylistEntity>
```
Рекомендация:
`Room` может возвращать `Flow`
Получится:
`fun getPlaylists(): Flow<List<PlaylistEntity>>`

Аналогично для других мест, где требуется возвращать `Flow`

IliaAnK commented on app/src/main/java/com/il76/playlistmaker/media/data/PlaylistInteractorImpl.kt:
```diff
@@ -0,0 +1,22 @@
+package com.il76.playlistmaker.media.data
```
Обязательно:
`Interactor` это сущность `domain` слоя. Нужно перенести в `domain`

**Аналогично** для `MediaPlayerInteractorImpl` и `SettingsInteractorImpl`.

IliaAnK commented:
> Я так и не смог нормально стилизовать поля ввода на экране создания плейлиста через состояния.

Пожалуйста, обратись за помощью к наставнику

Commit:
Реализации интеракторов перемещены в Domain слой.

Commit:
Добавил деструктивную миграцию для перехода с первой версии БД на последующие

Commit:
Перечитываем список плейлистов при добавлении трека в плейлист.
Потерянный фон тулбара.
Перекрасил кнопки в диалоговом окне в светлой теме.
Исправил миграцию (не была изменена версия БД)

Commit:
Клавиатура теперь не перекрывает поле ввода описания

Commit:
Скругление загруженной картинки

Commit:
Фон экрана добавления плейлиста

Commit:
Продолжаем эксперименты с цветами

Commit:
Убрал лишнюю кастомизацию и дублирование стилей.
Поля в неактивном состоянии серые, в активном синие.

il76 commented:
> > Я так и не смог нормально стилизовать поля ввода на экране создания плейлиста через состояния.
> 
> Пожалуйста, обратись за помощью к наставнику

Синим цветом надо выделять поле только если оно в фокусе. Если оно заполнено и не в фокусе – выделение не нужно
Вот ответ на аналогичный вопрос в "пачке".

il76 commented:
Надеюсь, остальное поправил...
